spec/requests/posts/update_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+describe 'PUT /posts/:id' do
+  subject { put "/posts/#{ post.to_param }", params; response }
+
+  let!(:post) { create :post }
+  let(:params) { {} }
+
+  context "when user is signed in" do
+    let!(:user) { create :user }
+
+    before { login_as user }
+
+    context "when post belongs to current user" do
+      let!(:post) { create :post, user: user }
+
+      context "when valid params given" do
+        let(:params) { { post: attributes_for(:post) } }
+
+        it { is_expected.to redirect_to(post) }
+      end
+
+      context "when invalid params given" do
+        let(:params) { { post: { title: "" } } }
+
+        it { is_expected.to be_unprocessable }
+
+        it { is_expected.to render_template 'edit' }
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.